crates/bevy_ecs/src/change_detection.rs

```diff
@@ -497,4 +508,35 @@ mod tests {
         assert_eq!(3, into_mut.ticks.last_change_tick);
         assert_eq!(4, into_mut.ticks.change_tick);
     }
+
+    #[test]
+    fn map_mut() {
+        use super::*;
+        struct Outer(i64);
+
+        let mut component_ticks = ComponentTicks {
+            added: 1,
+            changed: 2,
+        };
+        let ticks = Ticks {
+            component_ticks: &mut component_ticks,
+            last_change_tick: 2,
+            change_tick: 3,
+        };
+
+        let mut outer = Outer(0);
+        let ptr = Mut {
+            value: &mut outer,
+            ticks,
+        };
+        assert!(!ptr.is_changed());
+
+        // Perform a mapping operation.
+        let mut inner = ptr.map_unchanged(|x| &mut x.0);
+        assert!(!inner.is_changed());
+
+        // Mutate the inner value.
+        *inner = 64;
+        assert!(inner.is_changed());
```

Comment:
Just a question as I’m not super familiar with the change detection internals. Would this cause the outer container to also indicate a change?

I think it would be good to add another assertion to `outer` here to ensure it is/isn't changed.

Reply:
It shouldn't. I agree with the request for that assert though.